src/servers/http_server_v2.cc

```diff
@@ -0,0 +1,1539 @@
+
+  std::shared_ptr<TraceManager> trace_manager_;
+  std::shared_ptr<SharedMemoryManager> shm_manager_;
+
+  // The allocator that will be used to allocate buffers for the
+  // inference result tensors.
+  TRTSERVER_ResponseAllocator* allocator_;
+
+  re2::RE2 server_regex_;
+  re2::RE2 model_regex_;
+};
+
+TRTSERVER_Error*
+HTTPAPIServerV2::InferResponseAlloc(
+    TRTSERVER_ResponseAllocator* allocator, const char* tensor_name,
+    size_t byte_size, TRTSERVER_Memory_Type preferred_memory_type,
+    int64_t preferred_memory_type_id, void* userp, void** buffer,
+    void** buffer_userp, TRTSERVER_Memory_Type* actual_memory_type,
+    int64_t* actual_memory_type_id)
+{
+  AllocPayload* payload = reinterpret_cast<AllocPayload*>(userp);
+
+  evbuffer* evhttp_buffer = evbuffer_new();
+  if (evhttp_buffer == nullptr) {
+    return TRTSERVER_ErrorNew(
+        TRTSERVER_ERROR_INTERNAL,
+        std::string("failed to create evbuffer for output tensor").c_str());
+  } else {
+    payload->response_buffer_.push_back(evhttp_buffer);
+  }
+
+  const TensorShmMap* shm_map = payload->shm_map_;
+
+  *buffer = nullptr;
+  *buffer_userp = nullptr;
+  *actual_memory_type = preferred_memory_type;
+  *actual_memory_type_id = preferred_memory_type_id;
+
+  // Don't need to do anything if no memory was requested.
+  if (byte_size > 0) {
+    bool use_shm = false;
+
+    if (shm_map != nullptr) {
+      const auto& pr = shm_map->find(tensor_name);
+      if (pr != shm_map->end()) {
+        // If the output is in shared memory then check that the expected
+        // requested buffer size is at least the byte size of the output.
+        if (byte_size > pr->second.byte_size_) {
+          return TRTSERVER_ErrorNew(
+              TRTSERVER_ERROR_INTERNAL,
+              std::string(
+                  "expected buffer size to be at least " +
+                  std::to_string(pr->second.byte_size_) + " bytes but got " +
+                  std::to_string(byte_size) + " bytes in output tensor")
+                  .c_str());
+        }
+
+        *buffer = const_cast<void*>(pr->second.base_);
+        *actual_memory_type = pr->second.memory_type_;
+        *actual_memory_type_id = pr->second.device_id_;
+        use_shm = true;
+
+        LOG_VERBOSE(1) << "HTTP: using shared-memory for '" << tensor_name
+                       << "', size: " << byte_size << ", addr: " << *buffer;
+      }
+    }
+
+    if (!use_shm) {
+      // Can't allocate for any memory type other than CPU. If asked to
+      // allocate on GPU memory then force allocation on CPU instead.
+      if (*actual_memory_type != TRTSERVER_MEMORY_CPU) {
+        LOG_VERBOSE(1) << "HTTP: unable to provide '" << tensor_name << "' in "
+                       << MemoryTypeString(*actual_memory_type) << ", will use "
+                       << MemoryTypeString(TRTSERVER_MEMORY_CPU);
+        *actual_memory_type = TRTSERVER_MEMORY_CPU;
+        *actual_memory_type_id = 0;
+      }
+
+#if 0
```

Comment:
How can this be ifdeffed out? Did you start from the current version of http_server.cc? This code seems old.

Reply:
The support for shared memory is not added yet. It was ifdeffed out similar to the grpc_server_v2.cc [here](https://github.com/NVIDIA/tensorrt-inference-server/blob/master/src/servers/grpc_server_v2.cc#L1342)

Reply:
Isn't it inside the !use_shm conditional? But I think this code is old relative to some changes @GuanLuo made.

Reply:
I think @CoderHam `#if 0` the wrong place.. The code here is for the case that does NOT use shared memory. The code here seems to be up to date actually, each ResponseAlloc invocation creates and reserves space on separate evhttp buffer while previously it uses the one buffer for all.

Reply:
Fixed